ibis/client.py

```diff
@@ -211,12 +211,23 @@ def execute(self, expr, params=None, limit='default', **kwargs):
           Array expressions: pandas.Series
           Scalar expressions: Python scalar value
         """
+        import sqlalchemy
```

Comment:
this is interesting, we don't technically require sqlalchemy so i can see this might fail if we don't have it installed (of course we don't test this).

can you do this only on the SQL backends?

Reply:
This class is `SQLClient`, I assumed it was for the SQL backends, and `query_ast` was always a sqlalchemy object. But I think that was too much to assume. I'll restructure and move the sqlalchemy code to the `AlchemyClient` class. Thanks for pointing it out.